File: VINdecode/vinDecode.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Diagnostics;
using System.Net;
using System.IO;
using System.Xml.Linq;

namespace VINdecode
{
    class vinDecode
    {
        static void Main(string[] args)
        {
            String url = "http://ws.vinquery.com/restxml.aspx?accesscode=YOUR_ACCESS_CODE&reportType=YOUR_REPORT_TYPE&vin=YOUR_VIN";
            // YOUR_ACCESS_CODE: Your access code.
			      // YOUR_REPORT_TYPE: 0 - Basic, 1 - Standard, 2 - Extended, 3 - Lite
			      // YOUR_VIN: The vin you wish to decode.

            WebRequest request = WebRequest.Create(url);
            WebResponse response = request.GetResponse();

            // Getting and parsing the response.
            using (Stream dataStream = response.GetResponseStream())
            {
                StreamReader stream = new StreamReader(dataStream);
                String responseString = stream.ReadToEnd();

                XElement root = XElement.Parse(responseString);

                // Parsing:
                // Iterating over each instance of 'el' which is any element that is an 
                // Item tag with a Key attribute that is not null.
                // Where the Item tag is a child of the root tag, VIN tag, and Vehicle tag.
                IEnumerable<XElement> tests =
                    from el in root.Elements("VIN").Elements("Vehicle").Elements("Item")
                    where (string) el.Attribute("Key") != null
                    select el;
                foreach (XElement el in tests)
                    Console.WriteLine((string) el.Attribute("Key") + ": " + (string) el.Attribute("Value"));
            }
            
            response.Close();

        }
    }
}

File: VINdecode/vinDecode.java
package VINdecode;

import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

import javax.xml.namespace.QName;
import javax.xml.stream.XMLEventReader;
import javax.xml.stream.XMLInputFactory;
import javax.xml.stream.events.Attribute;
import javax.xml.stream.events.StartElement;
import javax.xml.stream.events.XMLEvent;

public class vinDecode {
	public static void main(String[] args) {
		try {
			final URL link = new URL(
			"http://ws.vinquery.com/restxml.aspx?accesscode=YOUR_ACCESS_CODE&reportType=YOUR_REPORT_TYPE&vin=YOUR_VIN");
			// YOUR_ACCESS_CODE: Your access code.
			// YOUR_REPORT_TYPE: 0 - Basic, 1 - Standard, 2 - Extended, 3 - Lite
			// YOUR_VIN: The vin you wish to decode.
			final HttpURLConnection con = (HttpURLConnection) link.openConnection();
			con.setRequestMethod("GET");

			parseXML(con);
			con.disconnect();

		} catch (Exception e) {
			e.printStackTrace();
		}
	}

	public static void parseXML(HttpURLConnection con) {
		try {
			// XML stream is parsed, assuming LITE data type request.
			XMLInputFactory factory = XMLInputFactory.newInstance();
			XMLEventReader eventReader = factory.createXMLEventReader(new InputStreamReader(con.getInputStream()));
			// Parsing response for model year, make, and model.
			// This can be edited to parse for any information required.
			while (eventReader.hasNext()) {
				String temp, print = "";
				try {
					while (eventReader.hasNext()) {
						XMLEvent event = eventReader.nextEvent();
						if (event.isStartElement()) {
							StartElement start = event.asStartElement();
							if (start.getName().getLocalPart() == "Item") {
								Attribute key = start.getAttributeByName(new QName("Key"));
								Attribute value = start.getAttributeByName(new QName("Value"));
								temp = key.getValue();
								switch (temp) {
									case "Model Year":
										print = "Model Year: " + value.getValue();
										break;
									case "Make":
										print += "Make: " + value.getValue();
										break;
									case "Model":
										print += "Model: " + value.getValue();
										break;
								}
								break;
							}
						}
					}

					System.out.println(print);

				} catch (Exception e) {
					e.printStackTrace();
				}
			}
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
}
